include/aws/crt/mqtt/Mqtt5Listener.h

```diff
@@ -0,0 +1,344 @@
+                 */
+                OnStoppedHandler onStopped;
+
+                /**
+                 * Callback handler triggered when client begins an attempt to connect to the remote endpoint.
+                 */
+                OnAttemptingConnectHandler onAttemptingConnect;
+
+                /**
+                 * Callback handler triggered when an MQTT PUBLISH packet is received by the client
+                 */
+                OnListenerPublishReceivedHandler onListenerPublishReceived;
+
+                /*
+                 * Reference to Mqtt5Client. As the listener has a dependency on mqtt5_client, we would like to keep the
+                 * Mqtt5Client alive.
+                 */
+                std::shared_ptr<Mqtt5Client> m_mqtt5Client;
+
+                /*
+                 * Self reference to keep self alive until the termination callback is invoked.
+                 */
+                std::shared_ptr<Mqtt5ListenerCore> m_selfReference;
+
+                /*
+                 * The callback flag used to indicate if it is safe to invoke the callbacks
+                 */
+                enum CallbackFlag
+                {
+                    INVOKE,
+                    IGNORE
+                } m_callbackFlag;
+
+                /*
+                 * Lock for the callbacks. This is used to protect the callback flag and callbacks.
+                 */
+                std::recursive_mutex m_callback_lock;
+
+                /**
+                 * Underlying c classes for data storage and operational members
+                 */
+                aws_mqtt5_listener *m_listener;
+
+                std::map<Crt::String, Crt::Mqtt5::OnPublishReceivedHandler> m_subscriptionMap;
+
+                Allocator *m_allocator;
+            };
+
+            /**
+             * An MQTT5 listener.
+             *
+             * This is a move-only type. Unless otherwise specified, all function arguments need only to live through
+             * the duration of the function call.
+             */
+            class AWS_CRT_CPP_API Mqtt5Listener final : public std::enable_shared_from_this<Mqtt5Listener>
+            {
+              public:
+                /**
+                 * Factory function for mqtt5 listener
+                 *
+                 * @param options: Mqtt5ListenerOption
+                 * @param client: Mqtt5Client to listen to
+                 * @param allocator allocator to use
+                 * @return a new mqtt5 listener
+                 */
+                static std::shared_ptr<Mqtt5Listener> NewMqtt5Listener(
+                    const Mqtt5ListenerOptions &options,
+                    const std::shared_ptr<Mqtt5Client> &client,
+                    Allocator *allocator = ApiAllocator()) noexcept;
+
+                /**
+                 * Get shared pointer of the Mqtt5ListenerCore. Mqtt5ListenerCore is inherited to
+                 * enable_shared_from_this to help with memory safety.
```

Comment:
Nitpick: I think you might want to change `inherited` to `used`? Since I think this class doesn't inherit or extend the class, but instead uses it for life cycle purposes. `utilized` might also work. Admittedly this is a VERY minor wording difference.